ultra/install_deps.sh

```diff
@@ -74,9 +81,95 @@ function do_install_for_macos {
+        if ! check_python_version_gte_3_7; then
+
+            echo "A >=3.7 python version not found"
+            read -p "Install python3.7? [Yn]" should_add_python_3_7
+            if [[ $should_add_python_3_7 =~ ^[yY\w]*$ ]]; then
+                echo ""
+                printf "This will run the following commands:\n$ sudo apt-get update\n$ sudo apt-get install software-properties-common\n$ sudo add-apt-repository ppa:deadsnakes/ppa\n$ sudo apt-get install python3.7 python3.7-dev python3.7-tk python3.7-venv"
+                echo ""
+                read -p "WARNING. Is this OK? If you are unsure choose no. [Yn]" should_add_python_3_7
+                # second check to make sure they really want to
+                if [[ $should_add_python_3_7 =~ ^[yY\w]*$ ]]; then
+                        sudo apt-get install software-properties-common
+                        sudo add-apt-repository ppa:deadsnakes/ppa
+                        sudo apt-get install python3.7 python3.7-dev python3.7-tk python3.7-venv
+                fi
+            fi
+        fi
+    elif [[ $UBUNTU_VERSION == "20.04" ]]; then
+        #only a problem for linux
+        if ! check_python_version_gte_3_8; then
+
+             echo "A >=3.8 python version not found"
+             read -p "Install python3.8? [Yn]" should_add_python_3_8
+             if [[ $should_add_python_3_8 =~ ^[yY\w]*$ ]]; then
+                  echo ""
+                  printf "This will run the following commands:\n$ sudo apt-get update\n$ sudo apt-get install software-properties-common\n$ sudo add-apt-repository ppa:deadsnakes/ppa\n$ sudo apt-get install python3.8 python3.8-dev python3.8-tk python3.8-venv"
+                  echo ""
+                  read -p "WARNING. Is this OK? If you are unsure choose no. [Yn]" should_add_python_3_8
+                  # second check to make sure they really want to
+                  if [[ $should_add_python_3_8 =~ ^[yY\w]*$ ]]; then
+                        sudo apt-get install software-properties-common
+                        sudo add-apt-repository ppa:deadsnakes/ppa
+                        sudo apt-get install python3.8 python3.8-dev python3.8-tk python3.8-venv
+                  fi
+             fi
+        fi
+    else
+        echo "Unsupported UBUNTU RELEASE: $UBUNTU_VERSION"
+        exit 1
+    fi
+
+    echo ""
+    echo "-- dependencies have been installed --"
+    echo ""
+    echo "You'll need to set the SUMO_HOME variable. Logging out and back in will"
+    echo "get you set up. Alternatively, in your current session, you can run:"
+    echo ""
+    echo "  source /etc/profile.d/sumo.sh"
+    echo ""
+}
+
 if [[ "$OSTYPE" == "linux-gnu" ]]; then
     echo "Detected Linux"
-    do_install_for_linux
+    # If kernel release is Microsoft-based then proceed with WSL
+    # installation otherwise native linux installation
+    DISTRIB=$(awk -F= '/^NAME/{print $2}' /etc/os-release)
+    if [[ $DISTRIB == '"Ubuntu"' ]]; then
```

Comment:
I wonder if this if statement should be in `do_install_for_WSL`? I think the way it is currently, a non-ubuntu distribution cannot natively install ULTRA (previously we didn't require the Linux distribution to be Ubuntu).